functions/Get-DbaDbFile.ps1

```diff
@@ -59,20 +68,54 @@ function Get-DbaDbFile {
         [Alias("Databases")]
         [object[]]$Database,
         [object[]]$ExcludeDatabase,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Database[]]$SmoDatabase,
         [object[]]$InputObject,
         [Alias('Silent')]
         [switch]$EnableException
     )
 
+    begin {
+
+        # Used to ensure parameter validation checks only run once.
```

Comment:
The issue with introducing `$initialParameterCheck = $true` is that it would be the first in the toolset and thus the behavior would be inconsistent. Please let it fail repeatedly.